### PR TITLE
Adds the RFC10 about the public offer

### PR DESCRIPTION
**Pull request by h4sh3d**
This is still work in progress and may be changed.

Adds RFC10 about the public offer content and serialization.

### CODE REVIEW

File: 10-public-offer.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+< [u8] len > < [u8; len] punish timeout value >
+< [u8] fee strategy > < [u8] len > < [u8; len] fixed or minimum value > (< [u8] len > < [u8; len] maximum value >)
+< [u8] future maker role >
+```
+
+### Amounts
+
+For Bitcoin, the amounts must be serialized as a eight bytes unsigned little endian integer representing the number of satoshi.
+
+For Monero, the amounts must be serialized as a eight bytes unsigned little endian integer representing the number of monerujo.
```
Comment (zkao) [inline 10-public-offer.md:132]: not atomic units? 1e-12 XMR

File: 10-public-offer.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+### Timelocks
+
+For Bitcoin, the timelocs values must be serialized as a four bytes unsigned little endian integer representing the `nSequence` field in the transaction and the number to push on the witness stack with a `CHECKSEQUENCEVERIFY` opcode.
```
Comment (zkao) [inline 10-public-offer.md:136]: timelocks*

**Review by zkao — APPROVED** [review]

**Comment by zkao** [comment]
Types that had to be encoded on farcaster-core PR #15

on that PR, I hash the string. 

`PDLEQ`
`Secp256k1`
`Ed25519`

File: 10-public-offer.md (hunk 1 above)
Comment (h4sh3d) [reply 10-public-offer.md:132]: You are right, 1e-12 is the atomic unit and is called piconero and not monerujo, will fix that

Comment (h4sh3d) [reply 10-public-offer.md]: Fixed in 2c1e4be82f010fc66a25e328e5f52a717cb26784

File: 10-public-offer.md (hunk 2 above)
Comment (h4sh3d) [reply 10-public-offer.md]: Fixed in 2c1e4be82f010fc66a25e328e5f52a717cb26784

**Comment by h4sh3d** [comment]
Bitcoin atomic unit is wrong, it's not 1e-10 but 1e-8

**Review by Lederstrumpf — APPROVED** [review]

**Pull request merged**